components/scream/src/physics/shoc/shoc_functions_f90.cpp

```diff
@@ -3037,10 +3037,51 @@ void pblintd_surf_temp_f(Int shcol, Int nlev, Int nlevi, Real* z, Real* ustar, R
 {
   // TODO
 }
+
 void pblintd_check_pblh_f(Int shcol, Int nlev, Int nlevi, Real* z, Real* ustar, bool* check, Real* pblh)
 {
-  // TODO
+  using SHOC         = Functions<Real, DefaultDevice>;
+  using Spack        = typename SHOC::Spack;
+  using Scalar       = typename SHOC::Scalar;
+  using Pack1        = typename ekat::Pack<Real, 1>;
+  using Bpack1       = typename ekat::Pack<bool, 1>;
+  using view_bool_1d = typename SHOC::view_1d<Bpack1>;
+  using view_1d      = typename SHOC::view_1d<Pack1>;
+  using view_2d      = typename SHOC::view_2d<Spack>;
+
+  std::vector<view_2d> views_2d(1);
+  ekat::host_to_device({z}, shcol, nlev, views_2d, true);
+  view_2d z_2d (views_2d[0]);
+
+  std::vector<view_1d> views_1d(2);
+  ekat::host_to_device({ustar, pblh}, shcol, views_1d);
+  view_1d ustar_1d (views_1d[0]),
+          pblh_1d  (views_1d[1]);
+
+  SHOC::view_1d<bool> check_1d("check", shcol);
+  const auto host_check_1d = Kokkos::create_mirror_view(check_1d);
+  for (auto k=0; k<shcol; ++k) {
+    host_check_1d(k) = check[k];
+  }
+  Kokkos::deep_copy(check_1d, host_check_1d);
```

Comment:
I think `host_to_device` can now handle `bool` for view scalar type?

```
std::vector<SHOC::view_1d<bool>> views_1d_bool(1);
ekat::host_to_device({check}, shcol, views_1d_bool);
SHOC::view_1d<bool> check_1d (views_1d_bool[0]);
```

Reply:
host_to_device and device_to_host support the pack data, we use view of bool, so it is not working with device_to_host.

Reply:
Could this be a `view_1d<Pack<bool,1>>` like we do with scalar 1d views? Again though, doesn't affect shoc_main so feel free to ignore.

Reply:
@xyuan , @tcclevenger is correct: a 1-size Pack would allow you to use host_to_device. Could you please try that?

Reply:
yes, it works. thanks reminding me.